internal/runtime/client/client.go

```diff
@@ -221,12 +229,15 @@ func (c *client) CallAllExtensions(ctx context.Context, hook runtimecatalog.Hook
 		}
 		// If the object namespace isn't matched by the registration NamespaceSelector skip the call.
 		if !namespaceMatches {
+			log.V(5).Info(fmt.Sprintf("skipping object %s/%s as it does not match selector %q of ExtensionConfig", forObject.GetNamespace(), forObject.GetName(), registration.NamespaceSelector))
 			continue
 		}
 
+		ctx := ctrl.LoggerInto(ctx, log)
```

Comment:
Let's do this directly after we modify the logger in l.196

It's very easy to miss that the ctx was propagated to other funcs in between and wasn't updated in the ctx. (e.g. matchNamespace, although we currently don't log in there).

It's just fail safer to update the ctx immediately after the logger from context was changed